---
title: "Cluster annotation"
format: 
  html:
    code-fold: true
editor: visual
toc: true
execute:
  warning: false
  message: false
---

# 1) load in the data,packages and custom functions

```{r, echo=FALSE}
easypackages::libraries("Seurat","tidyverse","SingleR", "ggridges", "viridis", "ggh4x", "celldex", "scCustomize", "SeuratWrappers", "ggExtra", "textTinyR", "patchwork", "pheatmap", "ggrepel", "tidyseurat", "ggpubr", "viridis", "writexl", "readxl", "presto", "scCustomize", "qs", "sctransform","glmGamPoi", "clusterProfiler", "enrichplot", "ExperimentHub", "scuttle", 'scmap',"cowplot",
                        'clusterProfiler',
                        'org.Mm.eg.db',
                        'tidyseurat',
                        'enrichplot',
                        'xlsx', 'clustree')
obj.v5 = read_rds("../../Documents/machiels_lab_viral/intermediate_data/seurat_obj_central.rds")
DefaultAssay(obj.v5) <- "RNA"


# The palette with grey:
cbp1 <- c("#999999", "#E69F00", "#56B4E9", "#009E73",
          "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

# The palette with black:
cbp2 <- c("#000000", "#E69F00", "#56B4E9", "#009E73",
          "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
```

#### This function takes a Seurat object or data frame (`seurat_tbl`) as input, calculates counts based on various factors, converts certain columns to factors, filters out specific conditions, calculates proportions of samples in a cluster, and then calculates z-scores of frequencies per cluster. Finally, it returns a modified data frame with these calculated values.

```{r}
make_zscore_and_freq_data <- function(seurat_tbl) {
  # Calculate counts of observations based on various factors
  # Parameters:
  #   seurat_tbl: A Seurat object or data frame containing the data
  # Returns:
  #   A modified data frame with z-scores and frequencies calculated
  
  seurat_tbl |> 
    dplyr::count(day_mock, sample_type, condition, sample_tag_ms4a3_pos_gabbr2, seurat_clusters, .drop = FALSE) |> 
    # Make columns factors
    mutate(
      day_mock = factor(day_mock, levels = c("Mock", "d8", "d60")),
      sample_type = factor(sample_type, levels = c("lung", "bal")),
      condition = as_factor(condition),
      sampletag_Ms4a3 = as_factor(sample_tag_ms4a3_pos_gabbr2),
      seurat_clusters = as_factor(seurat_clusters)
    ) |> 
    # Filter out specific conditions on Mock day
    filter(!(day_mock == "Mock" & (condition %in% c("PR8", "MuHV4", "PVM", "MAV1")))) |> 
    # Calculate proportions of samples in a given cluster
    group_by(day_mock, sample_type, condition, sample_tag_ms4a3_pos_gabbr2) |> 
    mutate(freq_cluster = n / sum(n)) |> 
    # Calculate z-score of frequencies per cluster
    group_by(seurat_clusters) |> 
    mutate(cluster_frequency_zscore = (freq_cluster - mean(freq_cluster)) / sd(freq_cluster))
}

```

#### This function takes a data frame (`x`) containing the necessary columns for plotting and creates a heatmap using ggplot2. It maps `condition` to the x-axis, `seurat_clusters` to the y-axis, and `cluster_frequency_zscore` to the fill color. The heatmap is then faceted based on `sample_tag_ms4a3_pos_gabbr2`, `sample_type`, and `day_mock`. Axis text on the x-axis is rotated by 45 degrees for better readability. Finally, the color gradient for the fill is set to blue (low values), white (mid values), and red (high values).

```{r}
frequency_heatmap <- function(x){ 
  x |>   ggplot(aes(condition,
                    seurat_clusters,
                    fill=cluster_frequency_zscore)) +
  geom_tile()+ 
  #scale_fill_viridis()+theme_bw() +
  ggh4x::facet_nested_wrap(
    vars(sample_tag_ms4a3_pos_gabbr2,
         sample_type,
         day_mock),
    nrow = 1,
    drop = TRUE,
    scales = "free_x"#,
    #space="free"
    )+
    theme_bw()+
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))+
    scale_fill_gradient2( low = "blue", mid = "white", high = "red")}
```

#### this function selects the n amount of top differentially expressed genes in descending order in terms of avg_log2 fold change

```{r}
select_top_genes <- function(df, n = 20) {
  top_genes <- by(df, df$cluster, function(cluster_df) {
    cluster_df[order(cluster_df$avg_log2FC, decreasing = TRUE), ][1:n, ]
  })
  do.call(rbind, top_genes)
}
```

# 2) Data exploration

## 2.1 UMAP plot full object

```{r}
obj.v5 |> DimPlot(group.by = "seurat_clusters")
```

## 2.2 Density plot showing embeddings of the different sample types

```{r}
#| fig-width: 18
#| fig-height: 10
obj.v5  |>
  ggplot(aes(umapharmony_1,umapharmony_2,
             color=day_mock_sample_type_mockTissueIncluded)) +
  geom_point(aes(color="cell")) +
  geom_density_2d(bins=20,linewidth=1) +
  scale_color_manual(values = cbp1 )+
  theme_bw() +
  ggtitle("local density of sample in umap embedding")+
  facet_wrap(~day_mock_sample_type_mockTissueIncluded)
ggsave("Density_tissueDays_clusters_originalIntegratedSeuratObject.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration" , width = 18, height = 10)
```

```{r}
#| fig-width: 18
#| fig-height: 18
samples <- obj.v5$day_mock_sample_type_mockTissueIncluded |> unique()
umap_sampletype_mock_point_plotlist <- list()
for (sample in samples) {
  obj.v5_filtered_for_sample <- obj.v5 |> as_tibble( ) |> filter(day_mock_sample_type_mockTissueIncluded==sample)
  obj.v5_not_sample <- obj.v5 |> as_tibble( ) |> filter(day_mock_sample_type_mockTissueIncluded!=sample)
  
  umap_sampletype_mock_point_plotlist[[sample]] <- obj.v5 |> 
    ggplot() +
    geom_point(data= obj.v5_not_sample,aes(umapharmony_1,umapharmony_2),color="grey")+
    geom_point(data= obj.v5_filtered_for_sample,aes(umapharmony_1,umapharmony_2),
               color="red",
               alpha=0.4,
               #plot points without outer border without alpha
               stroke = 0,
               shape=16)+
    theme_bw()+
    scale_fill_continuous(type = "viridis")+
    ggtitle(sample)+theme_bw()
}

ggarrange(plotlist = umap_sampletype_mock_point_plotlist)
ggsave("tissueDays_clusters_originalIntegratedSeuratObject.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration", width = 18, height = 10)
```

## 2.3 Density plot showing embeddings of the different Ms4a3 conditions

```{r}
#| fig-width: 12
#| fig-height: 10
samples <- obj.v5$sampletag_Ms4a3 |> unique()
umap_sampletag_Ms4a3_point_plotlist <- list()
for (sample in samples) {
  obj.v5_filtered_for_sample <- obj.v5 |> as_tibble( ) |> filter(sampletag_Ms4a3==sample)
  obj.v5_not_sample <- obj.v5 |> as_tibble( ) |> filter(sampletag_Ms4a3!=sample)
  
  umap_sampletag_Ms4a3_point_plotlist[[sample]] <- obj.v5 |> 
    ggplot() +
    geom_point(data= obj.v5_not_sample,aes(umapharmony_1,umapharmony_2),color="grey")+
    geom_point(data= obj.v5_filtered_for_sample,aes(umapharmony_1,umapharmony_2),
               color="red",
               alpha=0.4,
               #plot points without outer border without alpha
               stroke = 0,
               shape=16)+
    theme_bw()+
    scale_fill_continuous(type = "viridis")+
    ggtitle(sample)+theme_bw()
}

ggarrange(plotlist = umap_sampletag_Ms4a3_point_plotlist)
ggsave("Ms4a3Embeddings_originalIntegratedSeuratObject.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration")
```

```{r}
#| fig-width: 12
#| fig-height: 10
samples <- obj.v5$sampletag_Ms4a3 |> unique()
umap_sampletag_Ms4a3_dens_plotlist <- list()
for (sample in samples) {
  obj.v5_filtered_for_sample <- obj.v5 |> as_tibble( ) |> filter(sampletag_Ms4a3==sample)
  umap_sampletag_Ms4a3_dens_plotlist[[sample]] <- obj.v5 |> 
    ggplot(aes(umapharmony_1,umapharmony_2) ) +
    geom_point()+
    stat_density_2d(data= obj.v5_filtered_for_sample, 
                    aes(x= umapharmony_1,
                        y=umapharmony_2,
                        fill = after_stat(level)), geom = "polygon",contour_var = "ndensity"
                    )+
    theme_bw()+
    scale_fill_continuous(type = "viridis")+
    ggtitle(sample)+theme_bw()
}

#densities have to be adjusted, overlap
#https://stackoverflow.com/questions/76533721/plot-only-top-layers-of-ggplot-stat-density-2d-geom-density-2d-in-r

ggarrange(plotlist =umap_sampletag_Ms4a3_dens_plotlist)
ggsave("Density_Ms4a3Embeddings_originalIntegratedSeuratObject.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration")
```

## 2.4 Density plot showing embeddings of the different conditions

```{r}
#| fig-width: 18
#| fig-height: 18
samples <- obj.v5$condition |> unique()
umap_condition_plotlist <- list()
for (sample in samples) {
  obj.v5_filtered_for_sample <- obj.v5 |> as_tibble( ) |> filter(condition==sample)
  obj.v5_not_sample <- obj.v5 |> as_tibble( ) |> filter(condition!=sample)
  
  umap_condition_plotlist[[sample]] <- obj.v5 |> 
    ggplot() +
    geom_point(data= obj.v5_not_sample,aes(umapharmony_1,umapharmony_2),color="grey")+
    geom_point(data= obj.v5_filtered_for_sample,aes(umapharmony_1,umapharmony_2),
               color="red",
               alpha=0.4,
               #plot points without outer border without alpha
               stroke = 0,
               shape=16)+
    theme_bw()+
    scale_fill_continuous(type = "viridis")+
    ggtitle(sample)+theme_bw()
}

ggarrange(plotlist = umap_condition_plotlist)
ggsave("conditionsEmbeddings_originalIntegratedSeuratObject.png", path ="../../Documents/machiels_lab_viral/Case_study_annotation/exploration" , height = 18, width = 18)
```

## 2.5 Density plot showing embeddings of the different conditions and Ms4a3 together

```{r}
#| fig-width: 18
#| fig-height: 18
# samples <- obj.v5$sampletag_name 
# umap_sampletag_name_plotlist <- list()
# for (sample in samples) {
#   obj.v5_filtered_for_sample <- obj.v5 |> as_tibble( ) |> filter(sampletag_name==sample)
#   obj.v5_not_sample <- obj.v5 |> as_tibble( ) |> filter(sampletag_name!=sample)
#   
#   umap_sampletag_name_plotlist[[sample]] <- obj.v5 |> 
#     ggplot() +
#     geom_point(data= obj.v5_not_sample,
#                aes(umapharmony_1,umapharmony_2),
#                color="grey")+
#     geom_point(data= obj.v5_filtered_for_sample,
#                aes(umapharmony_1,umapharmony_2),
#                color="red",
#                alpha=0.4,
#                #plot points without outer border without alpha
#                stroke = 0,
#                shape=16)+
#     theme_bw()+
#     scale_fill_continuous(type = "viridis")+
#     ggtitle(sample)+theme_bw()
# }
# 
# ggarrange(plotlist = umap_sampletag_name_plotlist)
# ggsave("sample_Embeddings_originalIntegratedSeuratObject.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration", height = 18, width = 18)
```

## 2.6 Heatmaps for abseq data on expression of CD11c and SiglecF

```{r}
obj.v5 |>
  join_features(c("Siglecf-AbSeq", "Cd11c"),assay="adt", slot="scale.data")  |> 
group_by(.feature,seurat_clusters) |>
  summarise(mean_scaled_dsb=mean(.abundance_adt)) |>
  group_by(.feature) |> 
  mutate(zscore_mean_scaled_dsb=(mean_scaled_dsb-mean(mean_scaled_dsb))/sd(mean_scaled_dsb))|> na.omit() |> 
  mutate(zscore_mean_scaled_dsb=ifelse(zscore_mean_scaled_dsb>2,2,
                                       ifelse(zscore_mean_scaled_dsb<(-2),-2,zscore_mean_scaled_dsb))) |> 
  ggplot(aes(.feature,seurat_clusters, fill=zscore_mean_scaled_dsb)) +
  geom_tile()+ scale_fill_viridis() + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +ggtitle("mean dsb-normalized protein counts, z-score of cluster averages per marker")
ggsave("ADT_mean dsb-normalized protein counts, z-score of cluster averages per marker.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration")
```

## 2.6 Heatmaps for RNA data on alveolar markers

```{r}
obj.v5 |>
  join_features(c("Pparg", "Chil3", "Fabp5", "Flt1", "Fabp4", "Siglecf", "Car4", "Ear1", "Krt79", "Itgax", "SiglecF"),assay="RNA", slot="scale.data")  |> 
group_by(.feature,seurat_clusters) |>
  summarise(mean_scaled_dsb=mean(.abundance_RNA)) |>
  group_by(.feature) |> 
  mutate(zscore_mean_scaled_dsb=(mean_scaled_dsb-mean(mean_scaled_dsb))/sd(mean_scaled_dsb))|> na.omit() |> 
  mutate(zscore_mean_scaled_dsb=ifelse(zscore_mean_scaled_dsb>2,2,
                                       ifelse(zscore_mean_scaled_dsb<(-2),-2,zscore_mean_scaled_dsb))) |> 
  ggplot(aes(.feature,seurat_clusters, fill=zscore_mean_scaled_dsb)) +
  geom_tile()+ scale_fill_viridis() + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +ggtitle("mean dsb-normalized protein counts, z-score of cluster averages per marker")
ggsave("RNA_mean dsb-normalized protein counts, z-score of cluster averages per marker.png", path  = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration")
```

## 2.7 cluster frequencies per group

```{r}
#| fig-width: 18
#| fig-height: 18
p1 <- obj.v5 |>  as_tibble()  |> filter(sample_tag_ms4a3_pos_gabbr2!= "Gabbr2_pos_Ms4a3_neg") |>  make_zscore_and_freq_data() |>
  mutate(individual_sample=paste(day_mock,
                                 sample_type,
                                 condition,
                                 sample_tag_ms4a3_pos_gabbr2,
                                 sep = "_")) |> 
  ggplot(aes(individual_sample,freq_cluster,color=condition)) + geom_point()+   facet_wrap(~seurat_clusters, scales = "free", nrow=1) +theme(axis.text.x = element_blank(), axis.title = element_blank())
p1

p2 <- obj.v5 |>  as_tibble()  |> filter(sample_tag_ms4a3_pos_gabbr2!= "Gabbr2_pos_Ms4a3_neg") |>  make_zscore_and_freq_data() |>
  mutate(individual_sample=paste(day_mock,sample_type, condition,sample_tag_ms4a3_pos_gabbr2, sep = "_")) |> 
  ggplot(aes(individual_sample,freq_cluster,color=day_mock)) + geom_point()+  facet_wrap(~seurat_clusters, scales = "free", nrow=1)+ theme(axis.text.x = element_blank(), axis.title = element_blank())

p3 <- obj.v5 |>  as_tibble()  |> filter(sample_tag_ms4a3_pos_gabbr2!= "Gabbr2_pos_Ms4a3_neg") |>  make_zscore_and_freq_data() |>
  mutate(individual_sample=paste(day_mock,sample_type, condition,sample_tag_ms4a3_pos_gabbr2, sep = "_")) |> 
  ggplot(aes(individual_sample,freq_cluster,color=sample_type)) + geom_point()+
 facet_wrap(~seurat_clusters, scales = "free", nrow=1) +theme(axis.text.x = element_blank(), axis.title = element_blank())

p4 <- obj.v5 |>  as_tibble()  |> filter(sample_tag_ms4a3_pos_gabbr2!= "Gabbr2_pos_Ms4a3_neg") |>  make_zscore_and_freq_data() |>
  mutate(individual_sample=paste(day_mock,sample_type, condition,sample_tag_ms4a3_pos_gabbr2, sep = "_")) |> 
  ggplot(aes(individual_sample,freq_cluster,color=sample_tag_ms4a3_pos_gabbr2)) + geom_point()+scale_color_brewer()+ facet_wrap(~seurat_clusters, scales = "free", nrow=1, strip.position="bottom") + #x_axis_text_90+
   theme_classic()+
  theme(axis.text.x = element_blank(), axis.title = element_blank())

(p1 + theme(strip.text.x = element_blank()))/
  (p2+theme(strip.text.x = element_blank()))/
  (p3+theme(strip.text.x = element_blank()))/
  p4 +
  plot_layout(guides = 'collect')
ggsave(filename = "cluster frequencies per group.png",path ="../../Documents/machiels_lab_viral/Case_study_annotation/exploration" ,  height = 20, width = 30)
```

## 2.8 **Z-scores of cluster frequencies: heatmap**

```{r}
#| fig-width: 10
#| fig-height: 15
obj.v5 |> 
  as_tibble() |>  # Convert obj.v5 to a tibble
  separate(orig.ident, sep = "__", into = c("day", "sample_type"), remove = FALSE) |> 
  mutate(day=str_replace_all(day,c("viral.experiment.1"="d60",
                                   "viral.experiment.2"="d8") )) |> 
  mutate(day=ifelse(condition=="Mock",condition,day)) |> 
  mutate(day=factor(day, levels=c("Mock", "d8","d60")),
         sample_type=as_factor(sample_type),
         condition=as_factor(condition),
         sampletag_Ms4a3=as_factor(sampletag_Ms4a3),
         seurat_clusters=as_factor(seurat_clusters)
         ) |> 
  dplyr::group_by(day,sample_type, condition,sampletag_Ms4a3,seurat_clusters) |> 
  dplyr::count(day,sample_type, condition,sampletag_Ms4a3,seurat_clusters, .drop = FALSE) |> 
  mutate(sample=paste(day,sample_type,condition,sampletag_Ms4a3)) |> 
  group_by(sample) |> 
  filter(!(day=="Mock"&(condition %in%c("PR8","MuHV4", "PVM", "MAV1")))) |> 
  mutate(freq_cluster=n/sum(n)) |> 
  mutate(condition=as.character(condition)) |> 
  na.omit() |>  # This is where the error occurs, likely because the object isn't a data frame or tibble
  group_by(seurat_clusters) |> 
  mutate(cluster_mean_freq=mean(freq_cluster)) |> 
  mutate(cluster_mean_freq_zscore=(freq_cluster-cluster_mean_freq)/sd(freq_cluster)) |> 
  ggplot(aes(condition,seurat_clusters, fill=cluster_mean_freq_zscore)) +
  geom_tile()+ 
  ggh4x::facet_nested_wrap(
    vars(sampletag_Ms4a3,sample_type ,day), nrow = 1,drop = TRUE,scales = "free_x"
  )+theme_bw()+ theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))   + scale_fill_gradient2(  low = "blue",
  mid = "white",
  high = "red")+ggtitle("proportions of cells from a given sample distributed across clusters ",
                        "Ms4a3 classification as sorted")

ggsave(filename = "proportions of cells from a given sample distributed across clusters.png",path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration",  height = 20, width = 30, units = "cm")
```

## 2.9 dimplot conditions

```{r}
#| fig-width: 10
#| fig-height: 15
DimPlot_scCustom(obj.v5, split.by = "condition",colors_use = c("purple", "cyan", "lightgreen", "red", "orange", "blue", "pink", "green", "violet","grey", "lightblue", "darkgreen", "yellow", "gold", "magenta"))
ggsave("dimplot_virus.png", width = 25, height = 30, units = "cm", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/dimplot")
```

## 2.10 dimplot Ms4a3

```{r}
#| fig-width: 10
#| fig-height: 15
DimPlot_scCustom(obj.v5, split.by = "sampletag_Ms4a3",colors_use = c("purple", "cyan", "lightgreen", "red", "orange", "blue", "pink", "green", "violet","grey", "lightblue", "darkgreen", "yellow", "gold", "magenta"))
ggsave("dimplot_Ms4a3.png", width = 25, height = 30, units = "cm", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/dimplot")
```

## 2.11 dimplot tissue

```{r}
#| fig-width: 10
#| fig-height: 10
DimPlot_scCustom(obj.v5, split.by = "sample_type",colors_use = c("purple", "cyan", "lightgreen", "red", "orange", "blue", "pink", "green", "violet","grey", "lightblue", "darkgreen", "yellow", "gold", "magenta"))
ggsave("dimplot_tissue.png", width = 25, height = 30, units = "cm", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/dimplot")
```

## 2.12 dimplot day

```{r}
#| fig-width: 10
#| fig-height: 15
DimPlot_scCustom(obj.v5, split.by = "day_factor",colors_use = c("purple", "cyan", "lightgreen", "red", "orange", "blue", "pink", "green", "violet","grey", "lightblue", "darkgreen", "yellow", "gold", "magenta"))
ggsave("dimplot_day.png", width = 25, height = 30, units = "cm", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/dimplot")
```

## 2.13 dimplot tissue split by condition

```{r}
#| fig-width: 10
#| fig-height: 10
DimPlot_scCustom(obj.v5, split.by = "condition", group.by = "sample_type",colors_use = c("blue", "orange"))
ggsave("dimplot_virusGroupedTissue.png", width = 40, height = 30, units = "cm", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration" )
```

## 2.14 dimplot Ms4a3 split by condition

```{r}
#| fig-width: 10
#| fig-height: 10
DimPlot_scCustom(obj.v5, split.by = "condition", group.by = "sampletag_Ms4a3", colors_use = c("darkgrey", "red") )
ggsave("dimplot_virusGroupedMs4a3.png", width = 40, height = 30, units = "cm", path ="../../Documents/machiels_lab_viral/Case_study_annotation/exploration"  )
```

## 2.15 AbSeq exploration of H2-ab1 Itgax SiglecF and CD274 for Ms4a3 + -

```{r}
#| fig-width: 10
#| fig-height: 10
DefaultAssay(obj.v5) = "adt"
umap_am_alveolar_d60 <- list()
pallet = viridis(n = 10, option = "D")
genes_AM <- c("H2-ia-ie-AbSeq", "Cd11c", "Siglecf-AbSeq", "Cd274-AbSeq")

# Define a function to create and save the plot
create_and_save_plot <- function(gene) {
  plot <- FeaturePlot_scCustom(obj.v5, 
                              features = gene, 
                              label = TRUE, 
                              label.size = 5, 
                              split.by = "sampletag_Ms4a3", 
                              colors_use = pallet, 
                              label_feature_yaxis = FALSE)
  
  # Generate a unique filename based on the gene name
  filename <- paste0(gene, "_featureplot_splitby_Ms4a3.png")
  
  # Save the plot with specific path and dimensions
  ggsave(plot = plot, 
         filename = filename, 
         width = 30, 
         height = 20, 
         units = "cm", 
         path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/abseq/Ms4a3")
}

# Loop through genes and create/save plots
for (i in genes_AM) {
  create_and_save_plot(i)
}
```

## 2.16 AbSeq exploration of H2-ab1 Itgax SiglecF and CD274 for tissue

```{r}
#| fig-width: 10
#| fig-height: 10
DefaultAssay(obj.v5) = "adt"
umap_am_alveolar_d60 <- list()
genes_AM <- c("H2-ia-ie-AbSeq", "Cd11c", "Siglecf-AbSeq", "Cd274-AbSeq")

# Define a function to create and save the plot
create_and_save_plot <- function(gene) {
  plot <- FeaturePlot_scCustom(obj.v5, 
                              features = gene, 
                              label = TRUE, 
                              label.size = 5, 
                              split.by = "sample_type", 
                              label_feature_yaxis = FALSE)
  
  # Generate a unique filename based on the gene name
  filename <- paste0(gene, "_featureplot_splitby_tissue.png")
  
  # Save the plot with specific path and dimensions
  ggsave(plot = plot, 
         filename = filename, 
         width = 30, 
         height = 20, 
         units = "cm", 
         path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/abseq/tissue")
}

# Loop through genes and create/save plots
for (i in genes_AM) {
  create_and_save_plot(i)
}
```

## 2.17 AbSeq exploration of H2-ab1 Itgax SiglecF and CD274 for condition

```{r}
#| fig-width: 15
#| fig-height: 15
DefaultAssay(obj.v5) = "adt"
umap_am_alveolar_d60 <- list()
genes_AM <- c("H2-ia-ie-AbSeq", "Cd11c", "Siglecf-AbSeq", "Cd274-AbSeq")

# Define a function to create and save the plot
create_and_save_plot <- function(gene) {
  plot <- FeaturePlot_scCustom(obj.v5, 
                              features = gene, 
                              label = TRUE, 
                              label.size = 5, 
                              split.by = "condition", 
                              label_feature_yaxis = FALSE)
  
  # Generate a unique filename based on the gene name
  filename <- paste0(gene, "_featureplot_splitby_condition.png")
  
  # Save the plot with specific path and dimensions
  ggsave(plot = plot, 
         filename = filename, 
         width = 30, 
         height = 15, 
         units = "cm", 
         path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/abseq/condition")
}

# Loop through genes and create/save plots
for (i in genes_AM) {
  create_and_save_plot(i)
}
```

## 2.18 AbSeq exploration of H2-ab1 Itgax SiglecF and CD274 for day

```{r}
#| fig-width: 15
#| fig-height: 15
DefaultAssay(obj.v5) = "adt"
umap_am_alveolar_d60 <- list()
genes_AM <- c("H2-ia-ie-AbSeq", "Cd11c", "Siglecf-AbSeq", "Cd274-AbSeq")

# Define a function to create and save the plot
create_and_save_plot <- function(gene) {
  plot <- FeaturePlot_scCustom(obj.v5, 
                              features = gene, 
                              label = TRUE, 
                              label.size = 5, 
                              split.by = "day_factor", 
                              label_feature_yaxis = FALSE)
  
  # Generate a unique filename based on the gene name
  filename <- paste0(gene, "_featureplot_splitby_day.png")
  
  # Save the plot with specific path and dimensions
  ggsave(plot = plot, 
         filename = filename, 
         width = 30, 
         height = 15, 
         units = "cm", 
         path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/abseq/day")
}

# Loop through genes and create/save plots
for (i in genes_AM) {
  create_and_save_plot(i)
}
```

## 2.19 RNA exploration of H2-ab1 Itgax SiglecF and CD274 for Ms4a3 + -

```{r}
#| fig-width: 10
#| fig-height: 10
DefaultAssay(obj.v5) = "RNA"
umap_am_alveolar_d60 <- list()
pallet = viridis(n = 10, option = "D")
genes_AM <- c("H2-Ab1", "Itgax", "Pparg", "Siglecf", "Cd274")

# Define a function to create and save the plot
create_and_save_plot <- function(gene) {
  plot <- FeaturePlot_scCustom(obj.v5, 
                              features = gene, 
                              label = TRUE, 
                              label.size = 5, 
                              split.by = "sampletag_Ms4a3", 
                              colors_use = pallet, 
                              label_feature_yaxis = FALSE)
  
  # Generate a unique filename based on the gene name
  filename <- paste0(gene, "_featureplot_splitby_Ms4a3.png")
  
  # Save the plot with specific path and dimensions
  ggsave(plot = plot, 
         filename = filename, 
         width = 30, 
         height = 20, 
         units = "cm", 
         path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/RNA/Ms4a3")
}

# Loop through genes and create/save plots
for (i in genes_AM) {
  create_and_save_plot(i)
}
```

## 2.20 RNA exploration of H2-ab1 Itgax SiglecF and CD274 for tissue

```{r}
#| fig-width: 10
#| fig-height: 10
DefaultAssay(obj.v5) = "RNA"
umap_am_alveolar_d60 <- list()
pallet = viridis(n = 10, option = "D")
genes_AM <- c("H2-Ab1", "Itgax", "Pparg", "Siglecf", "Cd274")

# Define a function to create and save the plot
create_and_save_plot <- function(gene) {
  plot <- FeaturePlot_scCustom(obj.v5, 
                              features = gene, 
                              label = TRUE, 
                              label.size = 5, 
                              split.by = "sample_type", 
                              colors_use = pallet, 
                              label_feature_yaxis = FALSE)
  
  # Generate a unique filename based on the gene name
  filename <- paste0(gene, "_featureplot_splitby_tissue.png")
  
  # Save the plot with specific path and dimensions
  ggsave(plot = plot, 
         filename = filename, 
         width = 30, 
         height = 20, 
         units = "cm", 
         path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/RNA/tissue")
}

# Loop through genes and create/save plots
for (i in genes_AM) {
  create_and_save_plot(i)
}
```

## 2.21 RNA exploration of H2-ab1 Itgax SiglecF and CD274 for condition

```{r}
#| fig-width: 15
#| fig-height: 15
DefaultAssay(obj.v5) = "RNA"
umap_am_alveolar_d60 <- list()
pallet = viridis(n = 10, option = "D")
genes_AM <- c("H2-Ab1", "Itgax", "Pparg", "Siglecf", "Cd274")

# Define a function to create and save the plot
create_and_save_plot <- function(gene) {
  plot <- FeaturePlot_scCustom(obj.v5, 
                              features = gene, 
                              label = TRUE, 
                              label.size = 5, 
                              split.by = "condition", 
                              colors_use = pallet, 
                              label_feature_yaxis = FALSE)
  
  # Generate a unique filename based on the gene name
  filename <- paste0(gene, "_featureplot_splitby_condition.png")
  
  # Save the plot with specific path and dimensions
  ggsave(plot = plot, 
         filename = filename, 
         width = 40, 
         height = 20, 
         units = "cm", 
         path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/RNA/condition")
}

# Loop through genes and create/save plots
for (i in genes_AM) {
  create_and_save_plot(i)
}
```

## 2.21 RNA exploration of H2-ab1 Itgax SiglecF and CD274 for day

```{r}
#| fig-width: 15
#| fig-height: 15
DefaultAssay(obj.v5) = "RNA"
umap_am_alveolar_d60 <- list()
pallet = viridis(n = 10, option = "D")
genes_AM <- c("H2-Ab1", "Itgax", "Pparg", "Siglecf", "Cd274")

# Define a function to create and save the plot
create_and_save_plot <- function(gene) {
  plot <- FeaturePlot_scCustom(obj.v5, 
                              features = gene, 
                              label = TRUE, 
                              label.size = 5, 
                              split.by = "day_factor", 
                              colors_use = pallet, 
                              label_feature_yaxis = FALSE)
  
  # Generate a unique filename based on the gene name
  filename <- paste0(gene, "_featureplot_splitby_day.png")
  
  # Save the plot with specific path and dimensions
  ggsave(plot = plot, 
         filename = filename, 
         width = 30, 
         height = 20, 
         units = "cm", 
         path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/RNA/day")
}

# Loop through genes and create/save plots
for (i in genes_AM) {
  create_and_save_plot(i)
}
```

# 3) find markers for the clusters

```{r}
Idents(obj.v5) = "seurat_clusters"
markers_all <- FindAllMarkers(obj.v5,
                          assay = "RNA",
                          logfc.threshold = 0.5,
                          test.use = "wilcox",
                          min.pct = 0.2)
markers_all = select_top_genes(markers_all, n = 20)
write_xlsx(markers_all, path = "../../Documents/machiels_lab_viral/Case_study_annotation/markers/markers_v2.xlsx")
```

# 4) Cluster annotation with singleR with ImmGenData

This reference uses mouse bulk RNA-seq data from immunologic genome project

```{r}
DimPlot(obj.v5, reduction = "umap_harmony", label = TRUE)
```

sce \<- SingleCellExperiment(assays = list(counts = obj.v5\$data, logcounts = obj.v5\$scale.data))

```{r, echo =FALSE}
reference_igp = celldex::ImmGenData()
colData(reference_igp)

```

```{r}
#change the seurat object to a single cell experiment object for input in SingleR
sce = as.SingleCellExperiment(x = obj.v5, assay = c("RNA"))
# need log normalized counts as input for SingleR
sce = logNormCounts(sce)
# run SingleR
results_igp = SingleR(test = sce, ref = reference_igp, labels = reference_igp$label.main)
```

```{r}
# add singleR labels to the seurat object's metadata structure
obj.v5$singleR_igp = results_igp$labels
```

```{r}
#| fig-width: 10
#| fig-height: 15
DimPlot(obj.v5, reduction = "umap_harmony", group.by = "singleR_igp", label = TRUE)
ggsave("annotation_igp.png", width = 15, height = 10, units = "cm", path = "../../Documents/machiels_lab_viral/Case_study_annotation/singleR/igp")
```

```{r}
#| fig-width: 10
#| fig-height: 15
 DimPlot_scCustom(obj.v5, reduction = "umap_harmony", group.by = "singleR_igp", label = TRUE, split.by = "seurat_clusters", repel = TRUE, ggplot_default_colors = TRUE, num_columns = 3, split_seurat = TRUE)
ggsave("annotation_igp_per_cluster.png", width = 40, height = 50, units = "cm", path = "../../Documents/machiels_lab_viral/Case_study_annotation/singleR/igp")
```

# 5) Cluster annotation with singleR with tabula muris

scRNA seq database where we used the data from lung as reference aka tabula muris (<https://tabula-muris.ds.czbiohub.org/>)

```{r}
# load in the databas and select for the lung tissue data
EH <- ExperimentHub()
REF <- EH[['EH1617']]
REF <- REF[,!is.na(REF$cell_ontology_class)]
lung_ref <- REF[,REF$tissue == 'Lung']
# log normalize the counts as before 
lung_ref <- logNormCounts(lung_ref)
# run SingleR
results_tabulamuris <- SingleR(test = sce, ref = lung_ref, labels = lung_ref$cell_ontology_class)




```

```{r}
#| fig-width: 15
#| fig-height: 20
#visualize the celltypes of the different clusters
obj.v5$singleR_tabulamuris = results_tabulamuris$labels
DimPlot(obj.v5, reduction = "umap_harmony", group.by = "singleR_tabulamuris", label = TRUE)
ggsave("annotation_tabulamuris.png", width = 25, height = 20, units = "cm", path = "../../Documents/machiels_lab_viral/Case_study_annotation/singleR/tabulamuris")
```

```{r}
#| fig-width: 10
#| fig-height: 15
DimPlot_scCustom(obj.v5, reduction = "umap_harmony", group.by = "singleR_tabulamuris", label = TRUE, split.by = "seurat_clusters")
```

## 5.1 scMAP

```{r}
#| fig-width: 10
#| fig-height: 15

rowData(lung_ref)$feature_symbol <- rownames(lung_ref)
lung_ref <- lung_ref[!duplicated(rownames(lung_ref)), ]
lung_ref <- selectFeatures(lung_ref, suppress_plot = FALSE)

colData(lung_ref)$cell_type1 <- lung_ref$`cell_ontology_class`
lung_ref <- indexCluster(lung_ref)

rowData(sce)$feature_symbol <- rownames(sce)

SC.RES <- scmapCluster(projection = sce, index_list = list(TabulaMurisData_Lung = metadata(lung_ref)$scmap_cluster_index))

obj.v5$scmap_cluster_labs <- SC.RES
DimPlot(obj.v5, reduction = "umap_harmony", group.by = "scmap_cluster_labs", label = TRUE)
ggsave("annotation_tabulamuris_scmap.png", width = 25, height = 20, units = "cm", path = "../../Documents/machiels_lab_viral/Case_study_annotation/singleR/tabulamuris_scmap")
```

```{r}
obj.v5 |>  write_rds("../../Documents/machiels_lab_viral/intermediate_data/seurat_obj_central.rds")

```

# 6) Filtering of unwanted celltypes

```{r}
Idents(obj.v5) = "seurat_clusters"
obj.v5<- FindSubCluster(object = obj.v5, cluster = "1", graph.name = "RNA_snn", subcluster.name = "cluster_1_subclusters", resolution = 0.19, algorithm = 1)




obj.v5 <- FindSubCluster(object = obj.v5, cluster = "5", graph.name = "RNA_snn", resolution = 0.15, algorithm = 1, subcluster.name = "cluster_5_subclusters")



obj.v5 |> DimPlot(group.by = "cluster_1_subclusters", label=T) 
obj.v5 |> DimPlot(group.by = "cluster_5_subclusters", label=T) 



```

```{r}
# Extract metadata
metadata <- obj.v5@meta.data

# Replace the value 5 in 'cluster_1_subclusters' with values from 'cluster_5_subclusters' if they start with '5_'
metadata <- metadata %>%
  mutate(cluster_1_subclusters = if_else(cluster_1_subclusters == "5" & str_starts(cluster_5_subclusters, "5_"),
                                         cluster_5_subclusters,
                                         cluster_1_subclusters))

# Assign the modified metadata back to the Seurat object
obj.v5@meta.data <- metadata
```

```{r}
obj.v5$cluster_1_subclusters = factor(obj.v5$cluster_1_subclusters, levels = c("0", "1_0", "1_1", "1_2", "1_3", "2", "3", "4", "5_0", "5_1","5_2","5_3", "5_4", "5_5", "6", "7", "8","9", "10", "11", "12", "13", "14"))
obj.v5$seurat_clusters = obj.v5$cluster_1_subclusters

```

## 

```{r}
obj.v5 |> DimPlot(group.by = "seurat_clusters", label=T) 
ggsave("clusters_originalIntegratedSeuratObject_cluster1_recluster.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration", height = 12, width = 12)
```

## 6.1 plots checking expression of markers for celltypess

```{r}
Idents(obj.v5) = "seurat_clusters"
DotPlot_scCustom(seurat_object = obj.v5, features = c("Trac", "Cd4", "Cd8a", "Marco", "Siglecf"))
ggsave("dotplot_markers.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration", height = 12, width = 12, bg = "white")
```

```{r}
colors <- c(
  "#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00",
  "#FFFF33", "#A65628", "#F781BF", "#999999", "#66C2A5",
  "#FC8D62", "#8DA0CB", "#E78AC3", "#A6D854", "#FFD92F",
  "#E5C494", "#B3B3B3", "#1F78B4", "#33A02C", "#FB9A99",
  "#E31A1C", "#FDBF6F", "#CAB2D6"
)
Stacked_VlnPlot(seurat_object = obj.v5, features = c("Trac", "Cd4", "Cd8a", "Marco", "Siglecf"), x_lab_rotate = TRUE,
    colors_use = colors)
ggsave("violinplot_markers.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration", height = 12, width = 12)
```

```{r}
Idents(obj.v5)
obj.v5 <- subset(obj.v5, idents = c("5_1", "6","7","9", "10","11","12","13","14"), invert = TRUE)
```

## 6.2 UMAP plot filtered project

```{r}
obj.v5 |> DimPlot(group.by = "cluster_1_subclusters", label=T) 
ggsave("UMAP_subclustered_1_5_filtered.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration" , width = 10, height = 10)
```

# 7) Reclustering of filtered object

```{r, echo=FALSE}
#| fig-width: 10
#| fig-height: 15
obj.v5 <- FindNeighbors(obj.v5, reduction = "integrated.harmony", dims = 1:40)


# Find clusters using a range of resolutions
resolution.range <- seq(from = 0, to = 1, by = 0.2)

obj.v5 <- Seurat::FindClusters(object = obj.v5, resolution = resolution.range)
resolution.range <- seq(from = 0, to = 0.3, by = 0.02)

obj.v5 <- Seurat::FindClusters(object = obj.v5, resolution = resolution.range)
clustree(obj.v5)
ggsave("clustree_reclustering.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster" , width = 10, height = 10)

filtered_metadata <- obj.v5@meta.data[, !grepl("^RNA_snn", colnames(obj.v5@meta.data))]
obj.v5@meta.data <- filtered_metadata
print(colnames(obj.v5@meta.data))
```

```{r, echo=FALSE}
obj.v5 <- FindClusters(obj.v5, cluster.name = "harmony_clusters_0.2", resolution = 0.2)
obj.v5 <- RunUMAP(obj.v5, reduction = "integrated.harmony", reduction.name = "umap_harmony", dims = 1:40)
```

```{r}
obj.v5 |> DimPlot(group.by = "harmony_clusters_0.2", label=T)
ggsave("umap_reclustering.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster" , width = 10, height = 10)
```

```{r}
obj.v5 |> DimPlot(group.by = "seurat_clusters", label=T)
Idents(obj.v5) = "seurat_clusters"
```

# 8) Annotation of reclustered clusters with SingleR using the tabulamuris database

```{r}
sce = as.SingleCellExperiment(x = obj.v5, assay = c("RNA"))
# need log normalized counts as input for SingleR
sce = logNormCounts(sce)
```

```{r}
EH <- ExperimentHub()
REF <- EH[['EH1617']]
REF <- REF[,!is.na(REF$cell_ontology_class)]
lung_ref <- REF[,REF$tissue == 'Lung']
# log normalize the counts as before 
lung_ref <- logNormCounts(lung_ref)
# run SingleR
results_tabulamuris <- SingleR(test = sce, ref = lung_ref, labels = lung_ref$cell_ontology_class)
```

```{r}
#| fig-width: 15
#| fig-height: 20
#visualize the celltypes of the different clusters
obj.v5$singleR_tabulamuris = results_tabulamuris$labels
DimPlot(obj.v5, reduction = "umap_harmony", group.by = "singleR_tabulamuris", label = TRUE)
ggsave("umap_reclustering_annotated_tabulamuris.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster" , width = 10, height = 10)
```

```{r}
colors <- c(
  "#E41A1C", "#377EB8", "#4DAF4A", "#984EA3", "#FF7F00",
  "#FFFF33", "#A65628", "#F781BF", "#999999", "#66C2A5",
  "#FC8D62", "#8DA0CB", "#E78AC3", "#A6D854", "#FFD92F",
  "#E5C494", "#B3B3B3", "#1F78B4", "#33A02C", "#FB9A99",
  "#E31A1C", "#FDBF6F", "#CAB2D6"
)
Stacked_VlnPlot(seurat_object = obj.v5, features = c("Trac", "Cd4", "Cd8a", "Marco", "Siglecf", "Cd19", "Ptprc", "Ms4a1", "Cr2", "Ighm"), x_lab_rotate = TRUE,
    colors_use = colors)
ggsave("violinplot_markers.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster", height = 12, width = 12)
```

```{r}
Idents(obj.v5) = "singleR_tabulamuris"
obj.v5 <- subset(obj.v5, idents = c("T cell", "B cell", "natural killer cell", "stromal cell", "type II pneumocyte"), invert = TRUE)
```

```{r}
Idents(obj.v5) = "seurat_clusters"
obj.v5 |> DimPlot(group.by = "seurat_clusters", label=T)
ggsave("umap_reclustering_filteredforunwantedcells.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster" , width = 10, height = 10)
```

```{r}
DimPlot(obj.v5, reduction = "umap_harmony", group.by = "singleR_tabulamuris", label = TRUE)
ggsave("umap_reclustering_filteredforunwantedcells_annotated.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster" , width = 10, height = 10)
```

```{r}
Idents(obj.v5) = "seurat_clusters"
obj.v5<- FindSubCluster(object = obj.v5, cluster = "4", graph.name = "RNA_snn", subcluster.name = "cluster_4_subclusters", resolution = 0.19, algorithm = 1)
```

```{r}
obj.v5 |> DimPlot(group.by = "cluster_4_subclusters", label=T) 
```

```{r}
obj.v5<- FindSubCluster(object = obj.v5, cluster = "8", graph.name = "RNA_snn", subcluster.name = "cluster_8_subclusters", resolution = 0.19, algorithm = 1)
```

```{r}
obj.v5 |> DimPlot(group.by = "cluster_8_subclusters", label=T) 
```

```{r}
Idents(obj.v5) = "harmony_clusters_0.2"
# Extract metadata
metadata <- obj.v5@meta.data

# Replace the value 8 in 'cluster_4_subclusters' with values from 'cluster_8_subclusters' if they start with '8_'
metadata <- metadata %>%
  mutate(cluster_4_subclusters = if_else(cluster_4_subclusters == "8" & str_starts(cluster_8_subclusters, "8_"),
                                         cluster_8_subclusters,
                                         cluster_4_subclusters))

# Assign the modified metadata back to the Seurat object
obj.v5@meta.data <- metadata


```

```{r}
obj.v5$cluster_4_subclusters = factor(obj.v5$cluster_4_subclusters, levels = c("0", "1", "2", "3", "4_0", "4_1","4_2","4_3","4_4","5","6","7","8_0","8_1","8_2"))
obj.v5$seurat_clusters = obj.v5$cluster_4_subclusters
```

# 9) Data exploration reclustered + filtered

## 9.1 UMAP plot full object

```{r}

obj.v5 |> DimPlot(group.by = "seurat_clusters", label = TRUE)
ggsave("umap_reclustering_filteredforunwantedcells.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster" , width = 10, height = 10)
```

```{r}
obj.v5 |> DimPlot(group.by = "seurat_clusters", label = TRUE)
ggsave("umap_reclustering_filteredforunwantedcells_.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster" , width = 10, height = 10)
```

## 9.2 Density plot showing embeddings of the different sample types

```{r}
#| fig-width: 18
#| fig-height: 10
obj.v5  |>
  ggplot(aes(umapharmony_1,umapharmony_2,
             color=day_mock_sample_type_mockTissueIncluded)) +
  geom_point(aes(color="cell")) +
  geom_density_2d(bins=20,linewidth=1) +
  scale_color_manual(values = cbp1 )+
  theme_bw() +
  ggtitle("local density of sample in umap embedding")+
  facet_wrap(~day_mock_sample_type_mockTissueIncluded)
ggsave("Density_tissueDays_clusters_originalIntegratedSeuratObject.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration" , width = 18, height = 10)
```

```{r}
#| fig-width: 18
#| fig-height: 18
samples <- obj.v5$day_mock_sample_type_mockTissueIncluded |> unique()
umap_sampletype_mock_point_plotlist <- list()
for (sample in samples) {
  obj.v5_filtered_for_sample <- obj.v5 |> as_tibble( ) |> filter(day_mock_sample_type_mockTissueIncluded==sample)
  obj.v5_not_sample <- obj.v5 |> as_tibble( ) |> filter(day_mock_sample_type_mockTissueIncluded!=sample)
  
  umap_sampletype_mock_point_plotlist[[sample]] <- obj.v5 |> 
    ggplot() +
    geom_point(data= obj.v5_not_sample,aes(umapharmony_1,umapharmony_2),color="grey")+
    geom_point(data= obj.v5_filtered_for_sample,aes(umapharmony_1,umapharmony_2),
               color="red",
               alpha=0.4,
               #plot points without outer border without alpha
               stroke = 0,
               shape=16)+
    theme_bw()+
    scale_fill_continuous(type = "viridis")+
    ggtitle(sample)+theme_bw()
}

ggarrange(plotlist = umap_sampletype_mock_point_plotlist)
ggsave("tissueDays_clusters_originalIntegratedSeuratObject.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster", width = 18, height = 10)
```

## 9.3 Density plot showing embeddings of the different Ms4a3 conditions

```{r}
#| fig-width: 12
#| fig-height: 10
samples <- obj.v5$sampletag_Ms4a3 |> unique()
umap_sampletag_Ms4a3_point_plotlist <- list()
for (sample in samples) {
  obj.v5_filtered_for_sample <- obj.v5 |> as_tibble( ) |> filter(sampletag_Ms4a3==sample)
  obj.v5_not_sample <- obj.v5 |> as_tibble( ) |> filter(sampletag_Ms4a3!=sample)
  
  umap_sampletag_Ms4a3_point_plotlist[[sample]] <- obj.v5 |> 
    ggplot() +
    geom_point(data= obj.v5_not_sample,aes(umapharmony_1,umapharmony_2),color="grey")+
    geom_point(data= obj.v5_filtered_for_sample,aes(umapharmony_1,umapharmony_2),
               color="red",
               alpha=0.4,
               #plot points without outer border without alpha
               stroke = 0,
               shape=16)+
    theme_bw()+
    scale_fill_continuous(type = "viridis")+
    ggtitle(sample)+theme_bw()
}

ggarrange(plotlist = umap_sampletag_Ms4a3_point_plotlist)
ggsave("Ms4a3Embeddings_originalIntegratedSeuratObject.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster")
```

```{r}
#| fig-width: 12
#| fig-height: 10
samples <- obj.v5$sampletag_Ms4a3 |> unique()
umap_sampletag_Ms4a3_dens_plotlist <- list()
for (sample in samples) {
  obj.v5_filtered_for_sample <- obj.v5 |> as_tibble( ) |> filter(sampletag_Ms4a3==sample)
  umap_sampletag_Ms4a3_dens_plotlist[[sample]] <- obj.v5 |> 
    ggplot(aes(umapharmony_1,umapharmony_2) ) +
    geom_point()+
    stat_density_2d(data= obj.v5_filtered_for_sample, 
                    aes(x= umapharmony_1,
                        y=umapharmony_2,
                        fill = after_stat(level)), geom = "polygon",contour_var = "ndensity"
                    )+
    theme_bw()+
    scale_fill_continuous(type = "viridis")+
    ggtitle(sample)+theme_bw()
}

#densities have to be adjusted, overlap
#https://stackoverflow.com/questions/76533721/plot-only-top-layers-of-ggplot-stat-density-2d-geom-density-2d-in-r

ggarrange(plotlist =umap_sampletag_Ms4a3_dens_plotlist)
ggsave("Density_Ms4a3Embeddings_originalIntegratedSeuratObject.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster")
```

## 9.4 Density plot showing embeddings of the different conditions

```{r}
#| fig-width: 18
#| fig-height: 18
samples <- obj.v5$condition |> unique()
umap_condition_plotlist <- list()
for (sample in samples) {
  obj.v5_filtered_for_sample <- obj.v5 |> as_tibble( ) |> filter(condition==sample)
  obj.v5_not_sample <- obj.v5 |> as_tibble( ) |> filter(condition!=sample)
  
  umap_condition_plotlist[[sample]] <- obj.v5 |> 
    ggplot() +
    geom_point(data= obj.v5_not_sample,aes(umapharmony_1,umapharmony_2),color="grey")+
    geom_point(data= obj.v5_filtered_for_sample,aes(umapharmony_1,umapharmony_2),
               color="red",
               alpha=0.4,
               #plot points without outer border without alpha
               stroke = 0,
               shape=16)+
    theme_bw()+
    scale_fill_continuous(type = "viridis")+
    ggtitle(sample)+theme_bw()
}

ggarrange(plotlist = umap_condition_plotlist)
ggsave("conditionsEmbeddings_originalIntegratedSeuratObject.png", path ="../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster" , height = 18, width = 18)
```

## 9.5 Density plot showing embeddings of the different conditions and Ms4a3 together

```{r}
#| fig-width: 18
#| fig-height: 18
# samples <- obj.v5$sampletag_name 
# umap_sampletag_name_plotlist <- list()
# for (sample in samples) {
#   obj.v5_filtered_for_sample <- obj.v5 |> as_tibble( ) |> filter(sampletag_name==sample)
#   obj.v5_not_sample <- obj.v5 |> as_tibble( ) |> filter(sampletag_name!=sample)
#   
#   umap_sampletag_name_plotlist[[sample]] <- obj.v5 |> 
#     ggplot() +
#     geom_point(data= obj.v5_not_sample,
#                aes(umapharmony_1,umapharmony_2),
#                color="grey")+
#     geom_point(data= obj.v5_filtered_for_sample,
#                aes(umapharmony_1,umapharmony_2),
#                color="red",
#                alpha=0.4,
#                #plot points without outer border without alpha
#                stroke = 0,
#                shape=16)+
#     theme_bw()+
#     scale_fill_continuous(type = "viridis")+
#     ggtitle(sample)+theme_bw()
# }
# 
# ggarrange(plotlist = umap_sampletag_name_plotlist)
# ggsave("sample_Embeddings_originalIntegratedSeuratObject.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration", height = 18, width = 18)
```

## 9.6 Heatmaps for abseq data on expression of CD11c and SiglecF

```{r}
obj.v5 |>
  join_features(c("Siglecf-AbSeq", "Cd11c"),assay="adt", slot="scale.data")  |> 
group_by(.feature,seurat_clusters) |>
  summarise(mean_scaled_dsb=mean(.abundance_adt)) |>
  group_by(.feature) |> 
  mutate(zscore_mean_scaled_dsb=(mean_scaled_dsb-mean(mean_scaled_dsb))/sd(mean_scaled_dsb))|> na.omit() |> 
  mutate(zscore_mean_scaled_dsb=ifelse(zscore_mean_scaled_dsb>2,2,
                                       ifelse(zscore_mean_scaled_dsb<(-2),-2,zscore_mean_scaled_dsb))) |> 
  ggplot(aes(.feature,seurat_clusters, fill=zscore_mean_scaled_dsb)) +
  geom_tile()+ scale_fill_viridis() + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +ggtitle("mean dsb-normalized protein counts, z-score of cluster averages per marker")
ggsave("ADT_mean dsb-normalized protein counts, z-score of cluster averages per marker.png", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster")
```

## 9.7 Heatmaps for RNA data on alveolar markers

```{r}
obj.v5 |>
  join_features(c("Pparg", "Chil3", "Fabp5", "Flt1", "Fabp4", "Siglecf", "Car4", "Ear1", "Krt79", "Itgax", "SiglecF"),assay="RNA", slot="scale.data")  |> 
group_by(.feature,seurat_clusters) |>
  summarise(mean_scaled_dsb=mean(.abundance_RNA)) |>
  group_by(.feature) |> 
  mutate(zscore_mean_scaled_dsb=(mean_scaled_dsb-mean(mean_scaled_dsb))/sd(mean_scaled_dsb))|> na.omit() |> 
  mutate(zscore_mean_scaled_dsb=ifelse(zscore_mean_scaled_dsb>2,2,
                                       ifelse(zscore_mean_scaled_dsb<(-2),-2,zscore_mean_scaled_dsb))) |> 
  ggplot(aes(.feature,seurat_clusters, fill=zscore_mean_scaled_dsb)) +
  geom_tile()+ scale_fill_viridis() + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +ggtitle("mean dsb-normalized protein counts, z-score of cluster averages per marker")
ggsave("RNA_mean dsb-normalized protein counts, z-score of cluster averages per marker.png", path  = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster")
```

## 9.8 cluster frequencies per group

```{r}
#| fig-width: 18
#| fig-height: 18
p1 <- obj.v5 |>  as_tibble()  |> filter(sample_tag_ms4a3_pos_gabbr2!= "Gabbr2_pos_Ms4a3_neg") |>  make_zscore_and_freq_data() |>
  mutate(individual_sample=paste(day_mock,
                                 sample_type,
                                 condition,
                                 sample_tag_ms4a3_pos_gabbr2,
                                 sep = "_")) |> 
  ggplot(aes(individual_sample,freq_cluster,color=condition)) + geom_point()+   facet_wrap(~seurat_clusters, scales = "free", nrow=1) +theme(axis.text.x = element_blank(), axis.title = element_blank())
p1

p2 <- obj.v5 |>  as_tibble()  |> filter(sample_tag_ms4a3_pos_gabbr2!= "Gabbr2_pos_Ms4a3_neg") |>  make_zscore_and_freq_data() |>
  mutate(individual_sample=paste(day_mock,sample_type, condition,sample_tag_ms4a3_pos_gabbr2, sep = "_")) |> 
  ggplot(aes(individual_sample,freq_cluster,color=day_mock)) + geom_point()+  facet_wrap(~seurat_clusters, scales = "free", nrow=1)+ theme(axis.text.x = element_blank(), axis.title = element_blank())

p3 <- obj.v5 |>  as_tibble()  |> filter(sample_tag_ms4a3_pos_gabbr2!= "Gabbr2_pos_Ms4a3_neg") |>  make_zscore_and_freq_data() |>
  mutate(individual_sample=paste(day_mock,sample_type, condition,sample_tag_ms4a3_pos_gabbr2, sep = "_")) |> 
  ggplot(aes(individual_sample,freq_cluster,color=sample_type)) + geom_point()+
 facet_wrap(~seurat_clusters, scales = "free", nrow=1) +theme(axis.text.x = element_blank(), axis.title = element_blank())

p4 <- obj.v5 |>  as_tibble()  |> filter(sample_tag_ms4a3_pos_gabbr2!= "Gabbr2_pos_Ms4a3_neg") |>  make_zscore_and_freq_data() |>
  mutate(individual_sample=paste(day_mock,sample_type, condition,sample_tag_ms4a3_pos_gabbr2, sep = "_")) |> 
  ggplot(aes(individual_sample,freq_cluster,color=sample_tag_ms4a3_pos_gabbr2)) + geom_point()+scale_color_brewer()+ facet_wrap(~seurat_clusters, scales = "free", nrow=1, strip.position="bottom") + #x_axis_text_90+
   theme_classic()+
  theme(axis.text.x = element_blank(), axis.title = element_blank())

(p1 + theme(strip.text.x = element_blank()))/
  (p2+theme(strip.text.x = element_blank()))/
  (p3+theme(strip.text.x = element_blank()))/
  p4 +
  plot_layout(guides = 'collect')
ggsave(filename = "cluster frequencies per group.png",path ="../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster" ,  height = 20, width = 30)
```

## 9.9 **Z-scores of cluster frequencies: heatmap**

```{r}
#| fig-width: 10
#| fig-height: 15
obj.v5 |> 
  as_tibble() |>  # Convert obj.v5 to a tibble
  separate(orig.ident, sep = "__", into = c("day", "sample_type"), remove = FALSE) |> 
  mutate(day=str_replace_all(day,c("viral.experiment.1"="d60",
                                   "viral.experiment.2"="d8") )) |> 
  mutate(day=ifelse(condition=="Mock",condition,day)) |> 
  mutate(day=factor(day, levels=c("Mock", "d8","d60")),
         sample_type=as_factor(sample_type),
         condition=as_factor(condition),
         sampletag_Ms4a3=as_factor(sampletag_Ms4a3),
         seurat_clusters=as_factor(seurat_clusters)
         ) |> 
  dplyr::group_by(day,sample_type, condition,sampletag_Ms4a3,seurat_clusters) |> 
  dplyr::count(day,sample_type, condition,sampletag_Ms4a3,seurat_clusters, .drop = FALSE) |> 
  mutate(sample=paste(day,sample_type,condition,sampletag_Ms4a3)) |> 
  group_by(sample) |> 
  filter(!(day=="Mock"&(condition %in%c("PR8","MuHV4", "PVM", "MAV1")))) |> 
  mutate(freq_cluster=n/sum(n)) |> 
  mutate(condition=as.character(condition)) |> 
  na.omit() |>  # This is where the error occurs, likely because the object isn't a data frame or tibble
  group_by(seurat_clusters) |> 
  mutate(cluster_mean_freq=mean(freq_cluster)) |> 
  mutate(cluster_mean_freq_zscore=(freq_cluster-cluster_mean_freq)/sd(freq_cluster)) |> 
  ggplot(aes(condition,seurat_clusters, fill=cluster_mean_freq_zscore)) +
  geom_tile()+ 
  ggh4x::facet_nested_wrap(
    vars(sampletag_Ms4a3,sample_type ,day), nrow = 1,drop = TRUE,scales = "free_x"
  )+theme_bw()+ theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))   + scale_fill_gradient2(  low = "blue",
  mid = "white",
  high = "red")+ggtitle("proportions of cells from a given sample distributed across clusters ",
                        "Ms4a3 classification as sorted")

ggsave(filename = "proportions of cells from a given sample distributed across clusters.png",path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster",  height = 20, width = 30, units = "cm")
```

## 9.10 dimplot conditions

```{r}
#| fig-width: 10
#| fig-height: 15
DimPlot_scCustom(obj.v5, split.by = "condition",colors_use = c("purple", "cyan", "lightgreen", "red", "orange", "blue", "pink", "green", "violet","grey", "lightblue", "darkgreen", "yellow", "gold", "magenta"))
ggsave("dimplot_virus.png", width = 25, height = 30, units = "cm", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster/dimplot")
```

## 9.11 dimplot Ms4a3

```{r}
#| fig-width: 10
#| fig-height: 15
DimPlot_scCustom(obj.v5, split.by = "sampletag_Ms4a3",colors_use = c("purple", "cyan", "lightgreen", "red", "orange", "blue", "pink", "green", "violet","grey", "lightblue", "darkgreen", "yellow", "gold", "magenta"))
ggsave("dimplot_Ms4a3.png", width = 25, height = 30, units = "cm", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster/dimplot")
```

## 9.12 dimplot tissue

```{r}
#| fig-width: 10
#| fig-height: 10
DimPlot_scCustom(obj.v5, split.by = "sample_type",colors_use = c("purple", "cyan", "lightgreen", "red", "orange", "blue", "pink", "green", "violet","grey", "lightblue", "darkgreen", "yellow", "gold", "magenta"))
ggsave("dimplot_tissue.png", width = 25, height = 30, units = "cm", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster/dimplot")
```

## 9.13 dimplot day

```{r}
#| fig-width: 10
#| fig-height: 15
DimPlot_scCustom(obj.v5, split.by = "day_factor",colors_use = c("purple", "cyan", "lightgreen", "red", "orange", "blue", "pink", "green", "violet","grey", "lightblue", "darkgreen", "yellow", "gold", "magenta"))
ggsave("dimplot_day.png", width = 25, height = 30, units = "cm", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster/dimplot")
```

## 9.14 dimplot tissue split by condition

```{r}
#| fig-width: 10
#| fig-height: 10
DimPlot_scCustom(obj.v5, split.by = "condition", group.by = "sample_type",colors_use = c("blue", "orange"))
ggsave("dimplot_virusGroupedTissue.png", width = 40, height = 30, units = "cm", path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster/dimplot" )
```

## 9.15 dimplot Ms4a3 split by condition

```{r}
#| fig-width: 10
#| fig-height: 10
DimPlot_scCustom(obj.v5, split.by = "condition", group.by = "sampletag_Ms4a3", colors_use = c("darkgrey", "red") )
ggsave("dimplot_virusGroupedMs4a3.png", width = 40, height = 30, units = "cm", path ="../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster/dimplot"  )
```

## 9.16 AbSeq exploration of H2-ab1 Itgax SiglecF and CD274 for Ms4a3 + -

```{r}
#| fig-width: 10
#| fig-height: 10
DefaultAssay(obj.v5) = "adt"
umap_am_alveolar_d60 <- list()
pallet = viridis(n = 10, option = "D")
genes_AM <- c("H2-ia-ie-AbSeq", "Cd11c", "Siglecf-AbSeq", "Cd274-AbSeq")

# Define a function to create and save the plot
create_and_save_plot <- function(gene) {
  plot <- FeaturePlot_scCustom(obj.v5, 
                              features = gene, 
                              label = TRUE, 
                              label.size = 5, 
                              split.by = "sampletag_Ms4a3", 
                              colors_use = pallet, 
                              label_feature_yaxis = FALSE)
  
  # Generate a unique filename based on the gene name
  filename <- paste0(gene, "_featureplot_splitby_Ms4a3.png")
  
  # Save the plot with specific path and dimensions
  ggsave(plot = plot, 
         filename = filename, 
         width = 30, 
         height = 20, 
         units = "cm", 
         path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster/abseq/Ms4a3")
}

# Loop through genes and create/save plots
for (i in genes_AM) {
  create_and_save_plot(i)
}
```

## 9.17 AbSeq exploration of H2-ab1 Itgax SiglecF and CD274 for tissue

```{r}
#| fig-width: 10
#| fig-height: 10
DefaultAssay(obj.v5) = "adt"
umap_am_alveolar_d60 <- list()
genes_AM <- c("H2-ia-ie-AbSeq", "Cd11c", "Siglecf-AbSeq", "Cd274-AbSeq")

# Define a function to create and save the plot
create_and_save_plot <- function(gene) {
  plot <- FeaturePlot_scCustom(obj.v5, 
                              features = gene, 
                              label = TRUE, 
                              label.size = 5, 
                              split.by = "sample_type", 
                              label_feature_yaxis = FALSE)
  
  # Generate a unique filename based on the gene name
  filename <- paste0(gene, "_featureplot_splitby_tissue.png")
  
  # Save the plot with specific path and dimensions
  ggsave(plot = plot, 
         filename = filename, 
         width = 30, 
         height = 20, 
         units = "cm", 
         path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster/abseq/tissue")
}

# Loop through genes and create/save plots
for (i in genes_AM) {
  create_and_save_plot(i)
}
```

## 9.18 AbSeq exploration of H2-ab1 Itgax SiglecF and CD274 for condition

```{r}
#| fig-width: 15
#| fig-height: 15
DefaultAssay(obj.v5) = "adt"
umap_am_alveolar_d60 <- list()
genes_AM <- c("H2-ia-ie-AbSeq", "Cd11c", "Siglecf-AbSeq", "Cd274-AbSeq")

# Define a function to create and save the plot
create_and_save_plot <- function(gene) {
  plot <- FeaturePlot_scCustom(obj.v5, 
                              features = gene, 
                              label = TRUE, 
                              label.size = 5, 
                              split.by = "condition", 
                              label_feature_yaxis = FALSE)
  
  # Generate a unique filename based on the gene name
  filename <- paste0(gene, "_featureplot_splitby_condition.png")
  
  # Save the plot with specific path and dimensions
  ggsave(plot = plot, 
         filename = filename, 
         width = 30, 
         height = 15, 
         units = "cm", 
         path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster/abseq/condition")
}

# Loop through genes and create/save plots
for (i in genes_AM) {
  create_and_save_plot(i)
}
```

## 9.19 AbSeq exploration of H2-ab1 Itgax SiglecF and CD274 for day

```{r}
#| fig-width: 15
#| fig-height: 15
DefaultAssay(obj.v5) = "adt"
umap_am_alveolar_d60 <- list()
genes_AM <- c("H2-ia-ie-AbSeq", "Cd11c", "Siglecf-AbSeq", "Cd274-AbSeq")

# Define a function to create and save the plot
create_and_save_plot <- function(gene) {
  plot <- FeaturePlot_scCustom(obj.v5, 
                              features = gene, 
                              label = TRUE, 
                              label.size = 5, 
                              split.by = "day_factor", 
                              label_feature_yaxis = FALSE)
  
  # Generate a unique filename based on the gene name
  filename <- paste0(gene, "_featureplot_splitby_day.png")
  
  # Save the plot with specific path and dimensions
  ggsave(plot = plot, 
         filename = filename, 
         width = 30, 
         height = 15, 
         units = "cm", 
         path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster/abseq/day")
}

# Loop through genes and create/save plots
for (i in genes_AM) {
  create_and_save_plot(i)
}
```

## 9.20 RNA exploration of H2-ab1 Itgax SiglecF and CD274 for Ms4a3 + -

```{r}
#| fig-width: 10
#| fig-height: 10
DefaultAssay(obj.v5) = "RNA"
umap_am_alveolar_d60 <- list()
pallet = viridis(n = 10, option = "D")
genes_AM <- c("H2-Ab1", "Itgax", "Pparg", "Siglecf", "Cd274")

# Define a function to create and save the plot
create_and_save_plot <- function(gene) {
  plot <- FeaturePlot_scCustom(obj.v5, 
                              features = gene, 
                              label = TRUE, 
                              label.size = 5, 
                              split.by = "sampletag_Ms4a3", 
                              colors_use = pallet, 
                              label_feature_yaxis = FALSE)
  
  # Generate a unique filename based on the gene name
  filename <- paste0(gene, "_featureplot_splitby_Ms4a3.png")
  
  # Save the plot with specific path and dimensions
  ggsave(plot = plot, 
         filename = filename, 
         width = 30, 
         height = 20, 
         units = "cm", 
         path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster/RNA/Ms4a3")
}

# Loop through genes and create/save plots
for (i in genes_AM) {
  create_and_save_plot(i)
}
```

## 9.21 RNA exploration of H2-ab1 Itgax SiglecF and CD274 for tissue

```{r}
#| fig-width: 10
#| fig-height: 10
DefaultAssay(obj.v5) = "RNA"
umap_am_alveolar_d60 <- list()
pallet = viridis(n = 10, option = "D")
genes_AM <- c("H2-Ab1", "Itgax", "Pparg", "Siglecf", "Cd274")

# Define a function to create and save the plot
create_and_save_plot <- function(gene) {
  plot <- FeaturePlot_scCustom(obj.v5, 
                              features = gene, 
                              label = TRUE, 
                              label.size = 5, 
                              split.by = "sample_type", 
                              colors_use = pallet, 
                              label_feature_yaxis = FALSE)
  
  # Generate a unique filename based on the gene name
  filename <- paste0(gene, "_featureplot_splitby_tissue.png")
  
  # Save the plot with specific path and dimensions
  ggsave(plot = plot, 
         filename = filename, 
         width = 30, 
         height = 20, 
         units = "cm", 
         path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster/RNA/tissue")
}

# Loop through genes and create/save plots
for (i in genes_AM) {
  create_and_save_plot(i)
}
```

## 9.22 RNA exploration of H2-ab1 Itgax SiglecF and CD274 for condition

```{r}
#| fig-width: 15
#| fig-height: 15
DefaultAssay(obj.v5) = "RNA"
umap_am_alveolar_d60 <- list()
pallet = viridis(n = 10, option = "D")
genes_AM <- c("H2-Ab1", "Itgax", "Pparg", "Siglecf", "Cd274")

# Define a function to create and save the plot
create_and_save_plot <- function(gene) {
  plot <- FeaturePlot_scCustom(obj.v5, 
                              features = gene, 
                              label = TRUE, 
                              label.size = 5, 
                              split.by = "condition", 
                              colors_use = pallet, 
                              label_feature_yaxis = FALSE)
  
  # Generate a unique filename based on the gene name
  filename <- paste0(gene, "_featureplot_splitby_condition.png")
  
  # Save the plot with specific path and dimensions
  ggsave(plot = plot, 
         filename = filename, 
         width = 40, 
         height = 20, 
         units = "cm", 
         path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster/RNA/condition")
}

# Loop through genes and create/save plots
for (i in genes_AM) {
  create_and_save_plot(i)
}
```

## 9.23 RNA exploration of H2-ab1 Itgax SiglecF and CD274 for day

```{r}
#| fig-width: 15
#| fig-height: 15
DefaultAssay(obj.v5) = "RNA"
umap_am_alveolar_d60 <- list()
pallet = viridis(n = 10, option = "D")
genes_AM <- c("H2-Ab1", "Itgax", "Pparg", "Siglecf", "Cd274")

# Define a function to create and save the plot
create_and_save_plot <- function(gene) {
  plot <- FeaturePlot_scCustom(obj.v5, 
                              features = gene, 
                              label = TRUE, 
                              label.size = 5, 
                              split.by = "day_factor", 
                              colors_use = pallet, 
                              label_feature_yaxis = FALSE)
  
  # Generate a unique filename based on the gene name
  filename <- paste0(gene, "_featureplot_splitby_day.png")
  
  # Save the plot with specific path and dimensions
  ggsave(plot = plot, 
         filename = filename, 
         width = 30, 
         height = 20, 
         units = "cm", 
         path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster/RNA/day")
}

# Loop through genes and create/save plots
for (i in genes_AM) {
  create_and_save_plot(i)
}
```

# 10) find markers for the reclustered objects

```{r}
Idents(obj.v5) = "seurat_clusters"
Idents(obj.v5)
markers_all <- FindAllMarkers(obj.v5,
                          assay = "RNA",
                          logfc.threshold = 0.5,
                          test.use = "wilcox",
                          min.pct = 0.2)
markers_all = select_top_genes(markers_all, n = 20)
write_xlsx(markers_all, path = "../../Documents/machiels_lab_viral/Case_study_annotation/exploration/recluster/markers.xlsx")

```

```{r}
obj.v5 |>  write_rds("../../Documents/machiels_lab_viral/intermediate_data/seurat_obj_central_reclustered.rds")
```
